runtime/win32.c

```diff
@@ -1272,3 +1271,32 @@ value caml_win32_xdg_defaults(void)
 
   CAMLreturn(result);
 }
+
+static DWORD (WINAPI *get_temp_path)(DWORD, LPWSTR);
+static INIT_ONCE get_temp_path_init_once = INIT_ONCE_STATIC_INIT;
+static BOOL CALLBACK get_temp_path_init_function(PINIT_ONCE InitOnce,
+                                                 PVOID Parameter,
+                                                 PVOID *lpContext)
+{
+  get_temp_path = (DWORD (WINAPI *)(DWORD, LPWSTR))
+    GetProcAddress(GetModuleHandle(L"KERNEL32.DLL"), "GetTempPath2W");
+  if (!get_temp_path)
+    get_temp_path = GetTempPath;
+  return TRUE;
+}
+
+value caml_win32_get_temp_path(void)
+{
+  CAMLparam0();
+  CAMLlocal1(path);
+  wchar_t buf[MAX_PATH+1];
+
+  InitOnceExecuteOnce(&get_temp_path_init_once, get_temp_path_init_function,
+                      NULL, NULL);
```

Comment:
... retrieving it from the call here:
```suggestion
  wchar_t buf[MAX_PATH+1];
  DWORD (WINAPI *get_temp_path)(DWORD, LPWSTR);

  InitOnceExecuteOnce(&get_temp_path_init_once, get_temp_path_init_function,
                      NULL, &get_temp_path);
```